quickwit-doc-mapper/src/tokenizers.rs

```diff
@@ -17,14 +17,122 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+use std::str::CharIndices;
+
 use once_cell::sync::Lazy;
-use tantivy::tokenizer::{RawTokenizer, RemoveLongFilter, TextAnalyzer, TokenizerManager};
+use regex::Regex;
+use tantivy::tokenizer::{
+    BoxTokenStream, RawTokenizer, RemoveLongFilter, TextAnalyzer, Token, TokenStream, Tokenizer,
+    TokenizerManager,
+};
+
+static REGEX_ARRAY: [Lazy<Regex>; 3] = [
+    Lazy::new(|| Regex::new("^[a-z]+:?/?/?[a-z0-9]+[-_\\./a-z]*\\.[a-z]+[-/a-z._=]+").unwrap()),
+    Lazy::new(|| {
+        Regex::new(
+            "^[A-Za-z]{1,3}[ \t]{1,2}[0-9]{1,2}[ \
+             \t]{1,2}[0-9]{1,4}[-_/:][0-9]{1,2}[-_/:][0-9]{1,4}",
+        )
+        .unwrap()
+    }),
+    Lazy::new(|| Regex::new("^[0-9][-/%_\\.:a-zA-Z0-9]*").unwrap()),
+];
+static VALID_CHARS: &str = "%/-.";
+
+/// Log friendly Tokenizer that avoids splittings on IPs, dates, URLs and IDs
```

Comment:
This is too broad. You should explain which kind of IP are you targeting, which kind of dates. Moreover the order is important.